src/storage/txn/commands/flashback_to_version.rs

```diff
@@ -67,42 +70,58 @@ impl<S: Snapshot, L: LockManager> WriteCommand<S, L> for FlashbackToVersion {
             context.statistics,
         );
         let mut txn = MvccTxn::new(TimeStamp::zero(), context.concurrency_manager);
-
-        let mut next_lock_key = self.next_lock_key.take();
-        let mut next_write_key = self.next_write_key.take();
-        let rows = flashback_to_version(
-            &mut txn,
-            &mut reader,
-            &mut next_lock_key,
-            &mut next_write_key,
-            self.key_locks,
-            self.key_old_writes,
-            self.start_ts,
-            self.commit_ts,
-        )?;
+        // The state must be `ScanLock` or `ScanWrite` here.
+        match self.state {
+            FlashbackToVersionState::ScanLock {
+                ref mut next_lock_key,
+                ref key_locks,
+            } => {
+                if let Some(new_next_lock_key) =
+                    flashback_to_version_lock(&mut txn, &mut reader, key_locks)?
+                {
+                    *next_lock_key = Some(new_next_lock_key);
+                }
+            }
+            FlashbackToVersionState::ScanWrite {
+                ref mut next_write_key,
+                ref key_old_writes,
+            } => {
+                if let Some(new_next_write_key) = flashback_to_version_write(
+                    &mut txn,
+                    &mut reader,
+                    key_old_writes,
```

Comment:
Small suggestion (you can ignore it now if you're out of time before the freeze)

You can actually `take` this `Vec` out and let `flashback_to_version_write` accept `Vec` instead of `&Vec`. This will save some cloned.

And it's similar for the `key_locks` above. There are also similar cases in the read phase too.

Reply:
Resolved, PTAL.